src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -133,6 +133,30 @@ APPLICATION_INFO::CreateApplication(IHttpContext& pHttpContext)
 HRESULT
 APPLICATION_INFO::TryCreateApplication(IHttpContext& pHttpContext, const ShimOptions& options)
 {
+    const auto startupEvent = Environment::GetEnvironmentVariableValue(L"ASPNETCORE_STARTUP_SUSPEND_EVENT");
+    if (startupEvent.has_value())
+    {
+        LOG_INFOF(L"Startup suspend event %ls", startupEvent.value().c_str());
+
+        HandleWrapper<NullHandleTraits> eventHandle = OpenEvent(SYNCHRONIZE, false, startupEvent.value().c_str());
+
+        if (eventHandle == nullptr)
+        {
+            LOG_INFOF(L"Unable to open startup suspend event");
+        }
+        else
+        {
+            auto const suspendedEventName = startupEvent.value() + L"_suspended";
+
+            HandleWrapper<NullHandleTraits> suspendedEventHandle = OpenEvent(EVENT_MODIFY_STATE, false, suspendedEventName.c_str());
```

Comment:
Is this event only used for testing?

Reply:
Feels like we shouldn't open an event if it purely used for testing.

Reply:
Why not? Makes for better testing, caught a test bug on IIS. And I don't think it hurts anything.